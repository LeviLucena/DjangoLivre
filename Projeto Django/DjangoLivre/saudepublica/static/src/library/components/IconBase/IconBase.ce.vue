<template>
  <i :class="iconClasses" aria-hidden="true"></i>
</template>

<script>
const IconBase = {
  name: "IconBase",
  props: {
    /**
     * Nome do ícone que será apresentado
     */
    iconName: {
      type: String,
      default: "",
      validator: function (val) {
        return val.trim().length > 1
      },
    },
    /**
     * Se informado define que o nome da familia de fonte que será aplicada ao ícone
     */
    familyName: {
      type: String,
      default: "fas",
      validator: function (val) {
        return val.trim().length > 1
      },
    },
    /**
     * Se informado define qual o estilo de classe css que será aplicada ao ícone
     */
    className: {
      type: [String, Object],
      default: null,
    },
  },
  computed: {
    /**
     * Retorna o nome do ícone que será aplicado ao componente
     */
    iconClasses() {
      return [this.familyName, `fa-${this.processIconName(this.iconName)}`, this.className].join(" ")
    },
  },
  methods: {
    /**
     * Retorna o nome processado do ícone que será
     * apresentado
     */
    processIconName(iconName) {
      // Remove o prefixo "fa-" se estiver presente
      return iconName.replace(/^fa-/, "")
    },
  },
}

export default IconBase
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/_utilities";
@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css");
</style>
