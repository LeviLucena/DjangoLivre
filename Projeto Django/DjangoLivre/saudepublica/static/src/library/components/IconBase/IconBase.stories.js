import { kebabiseArgs } from '../../util/Utils.js'
import IconBase from './IconBase.ce.vue'

export default {
  title: 'Componentes/Icon',
  component: IconBase,
  argTypes: {
    iconName: {
      control: {
        type: 'text',
      },
    },
    familyName: {
      control: {
        type: 'text',
      },
    },
    className: {
      control: {
        type: 'text',
      },
    },
  },
}

const Template = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-icon-base v-bind="args"></br-icon-base>`,
})

export const Padrão = Template.bind({})
Padrão.args = {
  iconName: 'home',
  familyName: 'fas',
  className: '',
}
