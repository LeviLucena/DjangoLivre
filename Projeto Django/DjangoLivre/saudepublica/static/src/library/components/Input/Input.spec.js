import { mount, shallowMount } from '@vue/test-utils'
import { mask } from 'vue-the-mask'
import BrButton from '../Button/Button.ce.vue'
import IconBase from '../IconBase/IconBase.ce.vue'
import BrInput from './Input.ce.vue'

const densidades = ['large', 'medium', 'small']
const cores = ['info', 'warning', 'danger', 'success']

describe('Input', () => {
  test('find Input component and class "br-input"', () => {
    const wrapper = shallowMount(BrInput, {
      propsData: {},
    })
    expect(wrapper.classes('br-input')).toBe(true)
  })

  test('setValue on input', async () => {
    const wrapper = shallowMount(BrInput)
    const textInput = wrapper.find('input[type="text"]')
    await textInput.setValue('Inserindo texto no input')
    expect(wrapper.find('input[type="text"]').element.value).toBe('Inserindo texto no input')
  })

  test('setValue on input mask type cpf', async () => {
    const wrapper = shallowMount(BrInput, {
      propsData: {
        mask: '###.###.###-##',
      },
      directives: {
        mask,
      },
    })
    const textInput = wrapper.find('input[type="text"]')
    await textInput.setValue('614.523.270-68')
    await textInput.trigger('blur')
    expect(wrapper.find('input[type="text"]').element.value).toBe('614.523.270-68')
  })

  test('set label attribute', () => {
    const rotulo = 'Label do input'
    const wrapper = shallowMount(BrInput, {
      props: {
        label: rotulo,
      },
    })
    expect(wrapper.text()).toMatch(rotulo)
  })

  test('set label-inline attribute', () => {
    const rotulo = 'Label do input'
    const wrapper = shallowMount(BrInput, {
      props: {
        'label-inline': rotulo,
      },
    })
    expect(wrapper.text()).toMatch(rotulo)
    expect(wrapper.find('.input-inline').exists()).toBe(true)
    expect(wrapper.find('.input-group').exists()).toBe(false)
    const input = wrapper.findComponent('.br-input')
    expect(input.vm.isLabelInline).toBeTruthy()
    expect(wrapper.vm.isHighlight).toBe(false)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('set label-inline and icon-sign attributes', () => {
    const rotulo = 'Label do input'
    const wrapper = shallowMount(BrInput, {
      props: {
        'label-inline': rotulo,
        'icon-sign': 'edit',
      },
    })
    expect(wrapper.find('.input-group').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('set icon', () => {
    const icone = 'car'
    const wrapper = shallowMount(BrInput, {
      props: {
        icon: icone,
      },
    })

    expect(wrapper.classes()).toContain('br-input')
    expect(wrapper.classes()).toContain('input-button')
    expect(wrapper.props().icon).toBe(icone)
  })

  test('set is-highlight', () => {
    const wrapper = shallowMount(BrInput, {
      props: {
        isHighlight: true,
      },
    })
    expect(wrapper.find('.input-highlight').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('set iconSign', () => {
    const iconSign = 'user'
    const wrapper = shallowMount(BrInput, {
      props: {
        'icon-sign': iconSign,
      },
      global: {
        stubs: {
          'br-button': BrButton,
          'icon-base': IconBase,
        },
      },
    })

    const input = wrapper.findComponent('.br-input')
    expect(input.vm.hasIconsign).toBeTruthy()
    expect(wrapper.findComponent(IconBase).exists()).toBe(true)
  })

  test('has the expected html structure', () => {
    const wrapper = shallowMount(BrInput, {
      props: {},
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  densidades.forEach((densidade) => {
    test(`set density attribute ${densidade}`, () => {
      const wrapper = shallowMount(BrInput, {
        props: {
          density: densidade,
        },
      })
      expect(wrapper.find(`.${densidade}`).exists()).toBe(true)
    })
  })

  cores.forEach((cor) => {
    test(`set state attribute ${cor}`, () => {
      const wrapper = shallowMount(BrInput, {
        props: {
          state: cor,
        },
      })
      expect(wrapper.find(`.${cor}`).exists()).toBe(true)
    })
  })

  test("changing the element's value, updates the v-model", () => {
    const parent = mount({
      data() {
        return { login: '' }
      },
      methods: {
        logInput(e) {
          this.login = e
        },
        logChange(e) {
          this.login = e
        },
      },
      template: `<div>
        <br-input
          type="text"
          id="login"
          label="Login"
          v-model="login"
          @input="logInput"
          @change="logChange"
        >
        </br-input>
        </div>`,
      components: { BrInput },
    })

    const username = 'username_login'
    parent.find('input').setValue(username)

    expect(parent.vm.$data.login).toBe(username)
  })

  test('when is-password is passed as a prop, type must be "password"', () => {
    const wrapper = shallowMount(BrInput, {
      props: {
        ispassword: true,
      },
    })
    expect(wrapper.find('input[type="password"]').exists()).toBe(true)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('checks if the toggleIcon method is called', () => {
    const spyTogglePassword = jest.spyOn(BrInput.methods, 'toggleIcon')
    const wrapper = shallowMount(BrInput, {
      props: {
        ispassword: true,
      },
      global: {
        stubs: {
          'br-button': BrButton,
          'icon-base': IconBase,
        },
      },
    })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('.fas.fa-eye-slash').exists()).toBe(true)
    expect(wrapper.find('input[type="password"]').exists()).toBe(true)
    expect(wrapper.vm.$data.pswIcon).toBe('eye-slash')
    wrapper.find('.br-button').trigger('click')
    expect(wrapper.vm.toggleIcon).toBeTruthy()
    expect(spyTogglePassword).toHaveBeenCalled()
    wrapper.vm.$nextTick(() => {
      expect(wrapper.find('.fa-eye').exists()).toBe(true)
      expect(wrapper.find('input[type="text"]').exists()).toBe(true)
      expect(wrapper.vm.$data.pswIcon).toBe('eye')
      expect(wrapper.element).toMatchSnapshot()
    })
    spyTogglePassword.mockReset()
  })

  test('it renders inverted', () => {
    const rotulo = 'Label do input'
    const wrapper = shallowMount(BrInput, {
      props: {
        label: rotulo,
        inverted: true,
      },
    })
    expect(wrapper.classes('inverted')).toBe(true)
  })
})
