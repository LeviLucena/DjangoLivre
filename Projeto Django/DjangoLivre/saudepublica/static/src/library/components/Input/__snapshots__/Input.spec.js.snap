// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input checks if the toggleIcon method is called 1`] = `
<div
  class="br-input input-button"
>
  <!--v-if-->
  <!--v-if-->
  <!-- Input com máscaras -->
  <!--v-if-->
  <!-- Input sem máscaras -->
  <input
    id="br-input-44"
    type="password"
  />
  <!--v-if-->
  <button
    aria-label=""
    class="br-button small circle"
    type="button"
  >
    
    <i
      aria-hidden="true"
      class="fas fa-eye-slash"
    />
    <!--v-if-->
    
  </button>
  <!--v-if-->
</div>
`;

exports[`Input checks if the toggleIcon method is called 2`] = `
<div
  class="br-input input-button"
>
  <!--v-if-->
  <!--v-if-->
  <!-- Input com máscaras -->
  <!--v-if-->
  <!-- Input sem máscaras -->
  <input
    id="br-input-44"
    type="text"
  />
  <!--v-if-->
  <button
    aria-label=""
    class="br-button small circle"
    type="button"
  >
    
    <i
      aria-hidden="true"
      class="fas fa-eye"
    />
    <!--v-if-->
    
  </button>
  <!--v-if-->
</div>
`;

exports[`Input has the expected html structure 1`] = `
<div
  class="br-input"
>
  <!--v-if-->
  <!--v-if-->
  <!-- Input com máscaras -->
  <!--v-if-->
  <!-- Input sem máscaras -->
  <input
    id="br-input-22"
    type="text"
  />
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</div>
`;

exports[`Input set is-highlight 1`] = `
<div
  class="br-input input-highlight"
>
  <!--v-if-->
  <!--v-if-->
  <!-- Input com máscaras -->
  <!--v-if-->
  <!-- Input sem máscaras -->
  <input
    id="br-input-17"
    type="text"
  />
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</div>
`;

exports[`Input set label-inline and icon-sign attributes 1`] = `
<div
  class="br-input input-inline"
>
  <!--v-if-->
  <div
    class="input-label"
  >
    <label
      class="text-nowrap"
      for="br-input-11"
    >
      Label do input
    </label>
  </div>
  <!-- Input com máscaras -->
  <!--v-if-->
  <!-- Input sem máscaras -->
  <!--v-if-->
  <div
    class="input-content"
  >
    <div
      class="input-group"
    >
      <div
        class="input-icon"
      >
        <icon-base-stub
          class=""
          familyname="fas"
          iconname="edit"
        />
      </div>
      <input
        id="br-input-11"
        type="text"
      />
      <!--v-if-->
      <!--v-if-->
    </div>
  </div>
  <!--v-if-->
  <!--v-if-->
</div>
`;

exports[`Input set label-inline attribute 1`] = `
<div
  class="br-input input-inline"
>
  <!--v-if-->
  <div
    class="input-label"
  >
    <label
      class="text-nowrap"
      for="br-input-9"
    >
      Label do input
    </label>
  </div>
  <!-- Input com máscaras -->
  <!--v-if-->
  <!-- Input sem máscaras -->
  <div
    class="input-content"
  >
    <input
      id="br-input-9"
      type="text"
    />
  </div>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
</div>
`;

exports[`Input when is-password is passed as a prop, type must be "password" 1`] = `
<div
  class="br-input input-button"
>
  <!--v-if-->
  <!--v-if-->
  <!-- Input com máscaras -->
  <!--v-if-->
  <!-- Input sem máscaras -->
  <input
    id="br-input-41"
    type="password"
  />
  <!--v-if-->
  <br-button-stub
    active="false"
    block="false"
    circle="true"
    density="small"
    icon="eye-slash"
    inverted="false"
    loading="false"
  />
  <!--v-if-->
</div>
`;
